packages/runtime/src/helper.ts

```diff
@@ -880,6 +880,9 @@ export function getLanguageFromRequest(
     return defaultLanguage;
   }
 
+  // Copy the array so that it won't be mutated
+  appLanguages = [...appLanguages];
```

Comment:
What about combining this with the `unshift()` that caused all the trouble to start with?

```suggestion
  appLanguages = [defaultLanguage, ...appLanguages];
```